module/os/freebsd/zfs/zvol_os.c

```diff
@@ -839,6 +901,9 @@ zvol_cdev_read(struct cdev *dev, struct uio *uio_s, int ioflag)
 	    (zfs_uio_offset(&uio) < 0 || zfs_uio_offset(&uio) > volsize))
 		return (SET_ERROR(EIO));
 
+	if (!zvol_request_sync && !zv->zv_threading)
+		return (physread(dev, uio_s, ioflag));
```

Comment:
What's the point of this and the same for write below?  IIRC cdev read/write calls are always synchronous, so what are you achieving by this?

Reply:
It was needed because physio() creates bio from uio, the uio cannot be used in async [context](https://github.com/freebsd/freebsd-src/blob/main/sys/kern/subr_uio.c#L214).

Reply:
IIRC `d_read` and `d_write` methods are synchronous. It makes not sense to use threading for it, create bios, etc.

Reply:
The physio routines are removed.